src/main/java/io/fabric8/maven/docker/util/DockerFileUtil.java

```diff
@@ -166,10 +166,14 @@ public static FixedStringSearchInterpolator createInterpolator(MojoParameters pa
     }
 
     static String resolveArgValueFromStrContainingArgKey(String argString, Map<String, String> args) {
-        if (argString.startsWith("$") && args.containsKey(argString.substring(1))) {
-            return args.get(argString.substring(1));
-        } else if (argString.startsWith("${") && argString.endsWith("}") && args.containsKey(argString.substring(2, argString.length() - 1))) {
-            return args.get(argString.substring(2, argString.length() - 1));
+        Pattern argPattern = Pattern.compile("\\$(?:\\{(.*)\\}|(.*))");
+        Matcher matcher = argPattern.matcher(argString);
+        if (matcher.find()) {
```

Comment:
Should you use `matches()` here as find also matches on substrings ? (if you `^` and `$` this should make nor difference, but I still think "match" is cleare)